rcl_action/src/rcl_action/action_client.c

```diff
@@ -212,9 +214,22 @@ rcl_action_client_init(
 
   // Avoid uninitialized pointers should initialization fail
   *action_client->impl = _rcl_action_get_zero_initialized_client_impl();
+
+  // Remap/Expand the action name
+  char * resolved_action_name = NULL;
+  ret = rcl_node_resolve_name(node, action_name, allocator, false, false, &resolved_action_name);
+  if (RCL_RET_OK != ret) {
+    if (RCL_RET_TOPIC_NAME_INVALID == ret || RCL_RET_UNKNOWN_SUBSTITUTION == ret) {
+      ret = RCL_RET_ACTION_NAME_INVALID;
+    } else if (RCL_RET_BAD_ALLOC != ret) {
+      ret = RCL_RET_ERROR;
+    }
+    goto fail;
+  }
+
   // Copy action client name and options.
-  action_client->impl->action_name = rcutils_strdup(action_name, allocator);
-  if (NULL == action_client->impl->action_name) {
+  action_client->impl->remapped_action_name = rcutils_strdup(resolved_action_name, allocator);
+  if (NULL == action_client->impl->remapped_action_name) {
```

Comment:
this if statement does not make sense to me anymore, maybe we can remove this since this is for `rcutils_strdup` call which no longer required to copy.